catalog/spatial/registry/registry-report-viewer/src/main/java/org.codice.ddf.registry.report.viewer/RegistryReportViewer.java

```diff
@@ -0,0 +1,440 @@
+                    }
+                    String phoneNumber = countryCode + areaCode + number + extension + phoneType;
+                    phoneNumbers.add(phoneNumber);
+                }
+                organizationInfo.put("Phone Numbers", phoneNumbers);
+            }
+            if (organizationType.isSetEmailAddress()) {
+                List<String> emailAddresses = new ArrayList<>();
+                List<EmailAddressType> emailAddressTypes = organizationType.getEmailAddress();
+
+                for (EmailAddressType emailAddressType : emailAddressTypes) {
+                    emailAddresses.add(emailAddressType.getAddress());
+                }
+                organizationInfo.put("Email Addresses", emailAddresses);
+            }
+
+            List<SlotType1> slots = organizationType.getSlot();
+            for (SlotType1 slot : slots) {
+                String key = slot.getName();
+                List<String> values = slot.getValueList()
+                        .getValue()
+                        .getValue();
+                if (CollectionUtils.isEmpty(values)) {
+                    continue;
+                }
+                organizationInfo.put(key, StringUtils.join(values, ", "));
+            }
+
+            organizationMap.put(orgName, organizationInfo);
+        }
+
+        return organizationMap;
+    }
+
+    private Map<String, Object> getContactInfo(RegistryPackageType registryPackage) {
+        Map<String, Object> contactMap = new HashMap<>();
+        List<PersonType> persons = RegistryPackageUtils.getPersons(registryPackage);
+        for (PersonType person : persons) {
+            Map<String, Object> personMap = new HashMap<>();
+
+            PersonNameType personName = person.getPersonName();
+            String firstName = "";
+            String middleName = "";
+            String lastName = "";
+            if (personName.isSetFirstName()) {
+                firstName = personName.getFirstName() + " ";
+            }
+            if (personName.isSetMiddleName()) {
+                middleName = personName.getMiddleName() + " ";
+            }
+            if (personName.isSetLastName()) {
+                lastName = personName.getLastName();
+            }
+            String name = firstName + middleName + lastName;
+
+            if (person.isSetEmailAddress()) {
+                List<String> emailAddresses = new ArrayList<>();
+                List<EmailAddressType> emailAddressTypes = person.getEmailAddress();
+
+                for (EmailAddressType emailAddressType : emailAddressTypes) {
+                    emailAddresses.add(emailAddressType.getAddress());
+                }
+                personMap.put("Email Addresses", emailAddresses);
+            }
+
+            if (person.isSetTelephoneNumber()) {
+                List<String> phoneNumbers = new ArrayList<>();
+                List<TelephoneNumberType> telephoneNumberTypes = person.getTelephoneNumber();
```

Comment:
Looks like there is a lot of duplication for email/address/phone number parsing. could pull out into method to do this in one place.